lib/chef/provider/package.rb

```diff
@@ -170,6 +170,7 @@ def have_any_matching_version?
       end
 
       def removing_package?
+        validate_package(new_resource.package_name)
```

Comment:
we should probably validate on installing packages as well.

Reply:
(and update, and the rest of the APIs...)

Reply:
and really these validation calls should probably not be buried in the helper method but should go directly into `action_install` and all the rest of the top-class actions so its more readable.  having `removing_package?` throw an exception as a side effect is poor, even though it may be less code.

Reply:
Hi @lamont-granquist - Install and upgrade already checking and return `not found that package having space or invalid` that's why i have not added validation into that. Let me test if that required I'll add validation into that as well.

Reply:
define_resource_requirements is probably a better place to add the validation then.

you probably want to name the helper method something like `def valid_package_names?` and then subclasses can inherit and override that if the validation turns out to be not appropriate for some package manager out there, or if we want to add more specific validation to subclasses at some point.